vyper/parser/expr.py

```diff
@@ -328,8 +328,11 @@ def parse_Attribute(self):
             addr = Expr.parse_value_expr(self.expr.value, self.context)
             if is_base_type(addr.typ, "address"):
                 if self.expr.attr == "codesize":
-                    eval_code = ["extcodesize", addr]
-                    output_type = "int128"
+                    if self.expr.value.id == "self":
+                        eval_code = ["codesize"]
```

Comment:
This was added in Istanbul, right?

Reply:
Yes, `EXTCODEHASH` was added in Istanbul: https://eips.ethereum.org/EIPS/eip-1884

Reply:
This should be `... or evm_version < Istanbul`

And then add test cases for this behavior before/after Istanbul.

I think this feature was added recently, so maybe we missed this in the VIP?

Reply:
No.  You might be thinking of `EXTCODEHASH` that was added in constantinople?

Reply:
Ah, yes totally right. Disregard then.